### PR TITLE
[OC-49] Templates

### PR DESCRIPTION
**Pull request by nickbalestra**
Feature branch ready to be merged to master after
- [x] QA done on production ready hubs/components
- [x] QA on node4/6/8
CC @mattiaerre @matteofigus

### CODE REVIEW

**Comment by nickbalestra** [comment]
Please squash this

File: src/resources/index.js [hunk 1]
```diff
@@ -121,7 +121,7 @@ module.exports = {
       TEMPLATE_TYPE_NOT_VALID:
         'the template is not valid. Allowed values are handlebars and jade',
       TEMPLATE_DEP_MISSING:
-        'Template dependency missing. Run "$npm install --save {0}" to fix it.'
+        'Template dependency missing. To fix it run:\n\ncd {0} && npm install --save-dev {1}-compiler\n\n'
```
Comment (matteofigus) [inline src/resources/index.js]: Maybe we can just do `npm install --save-dev {1}-compiler --prefix {0}`. Perhaps reverting 0 with 1 for simplicity :D

File: test/unit/cli-domain-get-components-deps.js [hunk 2]
```diff
@@ -64,29 +70,32 @@ describe('cli : domain : get-components-deps', () => {
...
 
   describe('custom templates and package dependencies with versions', () => {
-
     before(() => {
       const readJsonStub = sinon.stub();
 
       readJsonStub.onFirstCall().returns({
         oc: baseOcFragment('oc-template-react'),
-        dependencies: { 'oc-template-react': '1.2.3'}
+        devDependencies: { 'oc-template-react-compiler': '1.2.3' }
```
Comment (matteofigus) [inline test/unit/cli-domain-get-components-deps.js]: can we add here `eslint` as devDependency too? So we can test that is not added to the ones that need to be installed

File: test/unit/cli-domain-package-components.js [hunk 3]
```diff
@@ -7,145 +7,94 @@ const sinon = require('sinon');
...
-        });
-
-        it('should save hash for template in package.json', () => {
-          expect(component.oc.files.template.hashKey).not.be.empty;
-        });
-
-        it('should minify static resources', () => {
-          expect(packageStaticFilesStub.args[0][0].minify).to.eql(false);
-        });
+    describe('when component is not valid', () => {
```
Comment (matteofigus) [inline test/unit/cli-domain-package-components.js]: When component parameters rename?

File: test/unit/utils-require-templates.js [hunk 4]
```diff
@@ -17,7 +17,7 @@ describe('utils : require-template', () => {
 
     scenarios.forEach(scenario => {
       it(scenario.name, () => {
```
Comment (matteofigus) [inline test/unit/utils-require-templates.js]: Let's add another little scenario for compiler true|false?

**Comment by nickbalestra** [comment]
Runned against our production apps and everything seemed to work correctly.
@matteofigus off to you for merging

**Pull request merged**